lib/driver.js

```diff
@@ -22,13 +21,20 @@ class AndroidDriver extends BaseDriver {
     [sessionId] = await super.createSession(caps);
 
     // TODO fail if app file is missing, or if app is not on the device?
-
+    // assigning defaults
+    let defaultOpts = {action: "android.intent.action.MAIN",
+                       category: "android.intent.category.LAUNCHER",
+                       flags: "0x10200000",
+                       disableAndroidWatchers: false};
+    _.defaults(this.opts, defaultOpts);
     if (!this.opts.javaVersion) {
       this.opts.javaVersion = await helpers.getJavaVersion();
     }
-
+    // Add condition for chrome driver
+    await this.checkAppPresent();
```

Comment:
I think there's an issue here.
Here we are checking for the presence of the app file, but at this time, driver.opts.app can be a url. We don't download or unzip apps until the `configureApp` function is called, in this case on line 63.
Maybe you can create an "check valid app cap" function which calls a `checkValidUrl` function if driver.opts.app is a valid url, otherwise call `checkAppPresent()`.